component/src/main/java/org/wso2/extension/siddhi/io/http/sink/util/HttpSinkUtil.java

```diff
@@ -128,42 +131,102 @@ public String trustStorePassword(ConfigReader sinkConfigReader) {
+                .get(HttpConstants.PORT));
         if (httpStaticProperties.get(HttpConstants.SCHEME).equals(HttpConstants.SCHEME_HTTPS)) {
-            SenderConfiguration httpsSender = new SenderConfiguration(httpStaticProperties
-                    .get(HttpConstants.PORT));
-            httpsSender.setTrustStoreFile(clientStoreFile);
-            httpsSender.setTrustStorePass(clientStorePass);
-            httpsSender.setId(httpStaticProperties.get(HttpConstants.TO));
-            httpsSender.setScheme(httpStaticProperties.get(HttpConstants.SCHEME));
-            senderConf = new HashSet<>(Collections.singletonList(httpsSender));
+            httpSender.setTrustStoreFile(clientStoreFile);
+            httpSender.setTrustStorePass(clientStorePass);
+            httpSender.setId(httpStaticProperties.get(HttpConstants.TO));
+            httpSender.setScheme(httpStaticProperties.get(HttpConstants.SCHEME));
+            //populateSenderStaticConfiguration(httpSender, configReader);
         } else {
-            SenderConfiguration httpSender = new SenderConfiguration(httpStaticProperties.get(HttpConstants.PORT));
             httpSender.setScheme(httpStaticProperties.get(HttpConstants.SCHEME));
-            senderConf = new HashSet<>(Collections.singletonList(httpSender));
+            // populateSenderStaticConfiguration(httpSender, configReader);
+
+        }
+        if (isHTTPTraceLoggerEnabled(configReader)) {
+            httpSender.setHttpTraceLogEnabled(true);
         }
-        return senderConf;
+        return httpSender;
+    }
+
+    private static boolean isHTTPTraceLoggerEnabled(ConfigReader configReader) {
+        return Boolean.parseBoolean(configReader.readConfig("httpTraceLogEnabled", "false"));
+    }
+
+    /**
+     * @param parameterList
+     * @return
+     */
+    public static List<org.wso2.transport.http.netty.config.Parameter> populateParameters(String parameterList) {
+        List<org.wso2.transport.http.netty.config.Parameter> parameters = new ArrayList<>();
+        if (!HttpConstants.EMPTY_STRING.equals(parameterList.trim())) {
+            try {
+                String[] valueList = parameterList.trim().substring(1,
+                        parameterList.length
+                                () - 1).split("','");
+                Arrays.stream(valueList).forEach(valueEntry ->
+                        {
+                            org.wso2.transport.http.netty.config.Parameter parameter = new Parameter();
+                            parameter.setName(valueEntry.split(":")[0]);
+                            parameter.setValue(valueEntry.split(":")[1]);
+                            parameters.add(parameter);
+                        }
+                );
+
+            } catch (ArrayIndexOutOfBoundsException ex) {
+                log.error("Bootstrap configuration is not in expected format please insert them as 'key1:val1'," +
+                        "'key2:val2' format");
```

Comment:
It might be useful to log the stacktrace as well.

Reply:
Fixed in 6bf095af9d7a4c00220069badb50517cec553139